File: Data Structures and Algorithms/[Book] Grokking Algorithms/Chapter 1/BinarySearch.java
public class BinarySearch{
    public static void main(String[] args){

        int myArray[] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16};
        int first = 0;
        int last = myArray.length;
        int guess = 10;


        while(first <= last){
            int mid = (first + last) / 2;

            if (myArray[mid] == guess){
                System.out.println(mid);
                break;
            }
            if (myArray[mid] > guess)
                    last = mid - 1;
            if (myArray[mid] < guess)
                first = mid + 1;
        }
    }
}

File: Data Structures and Algorithms/[Book] Grokking Algorithms/Chapter 1/binary_search.py

def binary_search(my_list: list, item: int) -> int:

    low = 0
    high = len(my_list) - 1
    while low <= high:
        mid = (low + high) // 2
        guess = my_list[mid]
        if guess < item:
            low = mid + 1
        if guess > item:
            high = mid - 1
        if guess == item:
            return mid
    return None


my_list = [1, 3, 5, 7, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
print(binary_search(my_list, 9))
print(binary_search(my_list, 7000))